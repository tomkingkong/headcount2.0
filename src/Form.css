.Form {
  display: flex;
  background-color: #2D2D2D;
  justify-content: center;
  align-items: center;
  height: 4rem;
  width: 100%;
  transition: all 0.45s linear;
}

.Form .search {
  border-radius: 0.2rem;
  color: #2d2d2d;
  margin: 0 auto;
  padding-left: 0.5rem;
  height: 2rem;
  width: 100%;
  max-width: 17rem;
  font-size: 1rem;
}