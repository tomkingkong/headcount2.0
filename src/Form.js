import React, { Component } from 'react';

import './Form.css'

class Form extends Component {
  constructor() {
    super();
    this.state = {
      input: ''
    }
  }

  handleChange = (e) => {
    const { value } = e.target;
    this.setState({
      input: value
    });
  }

  render() {
    return (
      <form 
        className="Form" 
        onSubmit={(e) => {
          e.preventDefault();
          
        }}
      >
        <input 
          type="text" 
          placeholder="Select a school"
          onChange={() => {
            this.handleChange;
            this.props.updateDistrict(this.state.input);
          }}
          value={this.state.input}
        />
      </form>
    );
  }
}

export default Form;