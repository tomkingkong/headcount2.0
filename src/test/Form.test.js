import React from 'react';
import { shallow } from 'enzyme';

import Form from '../Form';

describe('Form', () => {
  let wrapper;
  const updateDistrict = jest.fn();

  beforeEach(() => {
    wrapper = shallow(<Form updateDistrict={updateDistrict}/>);
  });

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should update state on change', () => {
    const event = { target: { value: 'col' } };
    expect(wrapper.state('input')).toEqual('');
    wrapper.instance().handleChange(event);
    expect(wrapper.state('input')).toEqual('col');
  });

  it('hanldeChange should update input value', () => {
    const event = { target: { value: 'col' } };
    wrapper.instance().handleChange(event);
    expect(wrapper.find('input').props().value).toEqual('col');
  });

  it('should call updateDistrict from props on change', () => {
    const event = { target: { value: 'col' } };
    wrapper.instance().handleChange(event);
    expect(updateDistrict).toHaveBeenCalled();
  });

  it('user should see the input they type on change', () => {
    const event = { target: { value: 'col' } };
    wrapper.find('input').simulate('change', event);
    expect(wrapper.find('input').props().value).toEqual('col');
  });

  it('should clear input on submit', () => {
    let event = { target: { value: 'col' } };
    wrapper.instance().handleChange(event);
    expect(wrapper.state('input')).toEqual('col');
    event = { preventDefault: jest.fn(), target: { value: 'col' } };
    wrapper.instance().handleSubmit(event);
    expect(wrapper.state('input')).toEqual('');
  });

  it('user should see form clear and repopulate on submit', () => {
    let event = { target: { value: 'col' } };
    wrapper.find('input').simulate('change', event);
    event = { preventDefault: jest.fn(), target: { value: 'col' } };
    wrapper.find('form').simulate('submit', event);
    expect(wrapper.find('input').props().value).toEqual('');
  });

});