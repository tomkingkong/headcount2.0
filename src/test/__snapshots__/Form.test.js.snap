// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Form
    updateDistrict={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <input
        className="search"
        onChange={[Function]}
        placeholder="Select card to compare.."
        type="text"
        value=""
/>,
      "className": "Form",
      "onSubmit": [Function],
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "className": "search",
        "onChange": [Function],
        "placeholder": "Select card to compare..",
        "type": "text",
        "value": "",
      },
      "ref": null,
      "rendered": null,
      "type": "input",
    },
    "type": "form",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <input
          className="search"
          onChange={[Function]}
          placeholder="Select card to compare.."
          type="text"
          value=""
/>,
        "className": "Form",
        "onSubmit": [Function],
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "search",
          "onChange": [Function],
          "placeholder": "Select card to compare..",
          "type": "text",
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      "type": "form",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "prevContext": true,
          },
        },
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
